NEWS

```diff
@@ -287,6 +287,12 @@ as part of https://github.com/networkupstools/nut/issues/1410 solution.
    would warn if a hostname resolves to several addresses (and would only
    listen on the first hit, as before in such cases) [#2012]
 
+ - A definitive behavior for `LISTEN *` directives became specified, to try
+   handling both IPv4 and IPv6 "any" address (subject to `upsd` CLI options
+   to only choose one, and to OS abilities); this use-case may be practically
+   implemented as a single IPv6 socket on systems with enabled "dual-stack"
```

Comment:
I don't think "dual-stack" is a proper term for "open a v6 socket and get v4 connections as mapped addresses".   To me dual stack just means that there is support for v4 and v6.   I would drop "; this use-case.*$" and add
```
`upsd` tries to open a v6 socket, asking not to use mapped addresses, and then a v4 socket.
```

Reply:
FWIW, "dual-stack" is what my initial reading-up named them. Probably stems from MSDN https://learn.microsoft.com/en-us/windows/win32/winsock/dual-stack-sockets which also mentions "IPv4-mapped IPv6 address format". I'd take it as a difference of technology (coded ability to support both) vs. format/representation of such addresses.

Also, for `LISTEN *` handling, this PR *does* ask to use mapped "ANY" addresses (unlike other cases with IPv6 where it asks not to). So it asks for IPv6, then IPv4, and depending on the OS "innate" abilities and its settings, gets one or two listening sockets to track.

Reply:
I can believe people use the term.  My point is that many people will read "dual stack" as "system supports v4 and v6" and not "system is configured so that when you listen on v6 it also listens on v4".   Hence asking that if what we are talking about is mapped addresses, to just say mapped addresses.

In 2023, a full OS (vs arduino code) that does not support IPv6 is deficient.  So I would also avoid dual stack on that basis, and instead flip the sense of the remarks to "As an exception, if the OS does not support  IPv6, or if this support was disabled when nut was configured.".   It's sort of like talking about cars that can use unleaded gas -- which used to be labeled as such and now isn't.

Reply:
Fair point :)